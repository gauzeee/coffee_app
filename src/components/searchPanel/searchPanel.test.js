import React from "react";
import SearchPanel from "./searchPanel";
import { shallow } from "enzyme";

describe("Testing SearchPanel", () => {
  const updateSearch = jest.fn();
  const updateFilter = jest.fn();
  const item = shallow(
    <SearchPanel updateSearch={updateSearch} updateFilter={updateFilter} />
  );

  describe("Testing SearchPanel snap and state", () => {
    it("SearchPanel have rendered correctly", () => {
      expect(item).toMatchSnapshot();
    });

    it("SearchPanel state 'searchText' is empty string", () => {
      expect(item.state().searchText).toEqual("");
    });
  });

  describe("SearchPanel handlers tests", () => {
    it("testing updateSearch", () => {
      item.instance().updateSearch({ target: { value: "test" } });
      expect(item.state().searchText).toEqual("test");
      expect(updateSearch).toBeCalledWith("test");
    });
    it("testing updateFilters", () => {
      item.instance().updateFilter({
        target: { value: "test", classList: { toggle: jest.fn() } }
      });
      expect(updateFilter).toBeCalled();
    });
  });
});
