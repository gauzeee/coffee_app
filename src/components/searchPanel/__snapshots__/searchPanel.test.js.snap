// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing SearchPanel Testing SearchPanel snap and state SearchPanel have rendered correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchPanel
    updateFilter={[MockFunction]}
    updateSearch={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="search-input-field"
        >
          <span>
            Looking for
          </span>
          <input
            onInput={[Function]}
            placeholder="start typing here..."
            type="text"
          />
        </div>,
        <div
          className="search-filters"
        >
          <span>
            Or filter
          </span>
          <button
            onClick={[Function]}
          >
            Brazil
          </button>
          <button
            onClick={[Function]}
          >
            Kenya
          </button>
          <button
            onClick={[Function]}
          >
            Columbia
          </button>
        </div>,
      ],
      "className": "search",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span>
              Looking for
            </span>,
            <input
              onInput={[Function]}
              placeholder="start typing here..."
              type="text"
            />,
          ],
          "className": "search-input-field",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Looking for",
            },
            "ref": null,
            "rendered": "Looking for",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "onInput": [Function],
              "placeholder": "start typing here...",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span>
              Or filter
            </span>,
            <button
              onClick={[Function]}
            >
              Brazil
            </button>,
            <button
              onClick={[Function]}
            >
              Kenya
            </button>,
            <button
              onClick={[Function]}
            >
              Columbia
            </button>,
          ],
          "className": "search-filters",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Or filter",
            },
            "ref": null,
            "rendered": "Or filter",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Brazil",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Brazil",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Kenya",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Kenya",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Columbia",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Columbia",
            "type": "button",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="search-input-field"
          >
            <span>
              Looking for
            </span>
            <input
              onInput={[Function]}
              placeholder="start typing here..."
              type="text"
            />
          </div>,
          <div
            className="search-filters"
          >
            <span>
              Or filter
            </span>
            <button
              onClick={[Function]}
            >
              Brazil
            </button>
            <button
              onClick={[Function]}
            >
              Kenya
            </button>
            <button
              onClick={[Function]}
            >
              Columbia
            </button>
          </div>,
        ],
        "className": "search",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span>
                Looking for
              </span>,
              <input
                onInput={[Function]}
                placeholder="start typing here..."
                type="text"
              />,
            ],
            "className": "search-input-field",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Looking for",
              },
              "ref": null,
              "rendered": "Looking for",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "onInput": [Function],
                "placeholder": "start typing here...",
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span>
                Or filter
              </span>,
              <button
                onClick={[Function]}
              >
                Brazil
              </button>,
              <button
                onClick={[Function]}
              >
                Kenya
              </button>,
              <button
                onClick={[Function]}
              >
                Columbia
              </button>,
            ],
            "className": "search-filters",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Or filter",
              },
              "ref": null,
              "rendered": "Or filter",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Brazil",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Brazil",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Kenya",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Kenya",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Columbia",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Columbia",
              "type": "button",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
